src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala

```diff
@@ -99,16 +94,38 @@ class RoutesPostgresRepo @Inject() (
   override def selectByName(name: String): DatabasePublisher[RouteRow] = {
     logger.debug(s"selectByName $name")
 
-    val q = for {
+    val q = (for {
       routeRow <- routesTable
       if routeRow.name === name && routeRow.deletedAt.isEmpty
-    } yield routeRow
+    } yield routeRow)
 
     db.stream {
       q.result
     }
   }
 
+  override def selectLatestActiveRoutesPerName(currentTime: LocalDateTime): DatabasePublisher[RouteRow] = {
+    logger.debug("select latest routes per name")
+
+    val q = (for {
+      routeRow <- routesTable
+      if routeRow.deletedAt.isEmpty && routeRow.activateAt < currentTime
```

Comment:
routes are immutable... when you create a route, you can specify an `activateAt` time, which can't be changed... if you want a route not to be valid anymore, you have to delete it...